src/data/viteComboFetcher.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * **WARNING** This module is expected to be invoked from a Vite Plugin!
+ * 
+ * Aggregated all YAML Files in the ./combos Directory and interpretes 
+ */
+
+import { parse } from "yaml"
+import { DbCombosTableZod, DbTricksTableZod } from "../lib/database/schemas/CurrentVersionSchema"
+import { readFile } from "node:fs/promises"
+import { ZodError, z } from "zod"
+import {globby} from "globby"
+import { join } from "node:path"
+import { fileURLToPath } from "node:url"
+import { YamlComboTableSchemaZod } from "./viteSchemaGenerator"
+
+async function fetchYamlFile(path: string): Promise<z.infer<typeof DbCombosTableZod>> {
+  const asObject = YamlComboTableSchemaZod.parse(parse(await readFile(path, "utf8")))
+
+  if(!asObject.tricks) {
+    throw new Error("Combos must contain at least one trick.")
+  }
+
+  asObject.tricks = asObject.tricks.map( trickId => ([trickId, "official"])) as any
+
+  // Makes sure the YAML has the right structure. 
+  return DbCombosTableZod.parse({...asObject, comboStatus: "official"})
+}
+
+function createRecordLookup<T extends z.infer<typeof DbCombosTableZod> | z.infer<typeof DbTricksTableZod>>(allEntities: T[]) {
+  let idToComboLookup: Record<number, T[]> = {};
+
+  allEntities.forEach(databaseEntity => {
+    if (!idToComboLookup[databaseEntity.id]) {
+      idToComboLookup[databaseEntity.id] = [];
+    }
+      idToComboLookup[databaseEntity.id].push(databaseEntity);
+    });
+  return idToComboLookup;
```

Comment:
I am confused a bit with the name `idToComboLookup` and that the input to the function are all official tricks and combos. Can you maybe explain a bit more on what it does?

Reply:
I should rename this to `idToEntityLookup`… function also should get some JSDoc…

but in short it creates a Map of Combo of Trick id => the Entity.

The result is then used to check how many Tricks where are with that ID. (used in `findDuplicateKeys` and `findUndefinedTrickReferences`)

Reply:
Here's the JSDoc I ill be adding

```ts
/**
 * This takes a list of Database Entities and creates a mapping of ID -> List of Objects
 * 
 * This can be used to detect duplicates or to check if an Entity with a given ID exists.
 */
```